src/app/clusters/scenes/SceneTable.h

```diff
@@ -0,0 +1,293 @@
+    /// @param cluster Target Cluster
+    /// @param serialisedBytes Output buffer, data needs to be writen in there and size adjusted if smaller than
+    /// kMaxFieldsPerCluster
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR value otherwise
+    virtual CHIP_ERROR SerializeSave(EndpointId endpoint, ClusterId cluster, MutableByteSpan & serialisedBytes) = 0;
+
+    /// @brief From stored scene (e.g. ViewScene), deserialize ExtensionFieldSet into a cluster object
+    /// @param endpoint Endpoint ID
+    /// @param cluster Cluster ID to save
+    /// @param serialisedBytes ExtensionFieldSet stored in NVM
+    /// @param extensionFieldSet ExtensionFieldSet in command format
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR value otherwise
+    virtual CHIP_ERROR Deserialize(EndpointId endpoint, ClusterId cluster, ByteSpan & serialisedBytes,
+                                   app::Clusters::Scenes::Structs::ExtensionFieldSet::Type & extensionFieldSet) = 0;
+
+    /// @brief From stored scene (e.g RecallScene), applies EFS values to cluster at transition time
+    /// @param endpoint Endpoint ID
+    /// @param cluster Cluster ID
+    /// @param serialisedBytes ExtensionFieldSet stored in NVM
+    /// @param timeMs Transition time in ms to apply the scene
+    /// @return CHIP_NO_ERROR if successful, CHIP_ERROR value otherwise
+    virtual CHIP_ERROR ApplyScene(EndpointId endpoint, ClusterId cluster, ByteSpan & serialisedBytes, TransitionTimeMs timeMs) = 0;
+};
+
+template <class EFStype>
+class SceneTable
+{
+public:
+    /// @brief struct used to identify a scene in storage by 3 ids, endpoint, group and scene
+    struct SceneStorageId
+    {
+        // Identifies endpoint to which this scene applies
+        EndpointId mEndpointId = kInvalidEndpointId;
+        // Identifies group within the scope of the given fabric
+        SceneGroupID mGroupId = kGlobalGroupSceneId;
+        SceneId mSceneId      = kUndefinedSceneId;
+
+        SceneStorageId() = default;
+        SceneStorageId(EndpointId endpoint, SceneId id, SceneGroupID groupId = kGlobalGroupSceneId) :
+            mEndpointId(endpoint), mGroupId(groupId), mSceneId(id)
+        {}
+
+        void Clear()
+        {
+            mEndpointId = kInvalidEndpointId;
+            mGroupId    = kGlobalGroupSceneId;
+            mSceneId    = kUndefinedSceneId;
+        }
+
+        bool operator==(const SceneStorageId & other)
+        {
+            return (this->mEndpointId == other.mEndpointId && this->mGroupId == other.mGroupId && this->mSceneId == other.mSceneId);
+        }
+    };
+
+    /// @brief struct used to store data held in a scene
+    struct SceneData
+    {
+        char mName[kSceneNameMax]                       = { 0 };
+        size_t mNameLength                              = 0;
+        SceneTransitionTime mSceneTransitionTimeSeconds = 0;
+        EFStype mExtensionFieldSets;
+        TransitionTime100ms mTransitionTime100ms = 0;

```

Comment:
It's a little confusing that we are storing two different transition times here.  Please document why.